### PR TITLE
gitlab_project: add support for merge_method on projects

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY

Migrated from ansible/ansible#66813

Fixes: https://github.com/ansible/ansible/issues/66812

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
changelogs/fragments/66813_gitlab_project.yml
plugins/modules/source_control/gitlab/gitlab_project.py
tests/unit/plugins/modules/source_control/gitlab/gitlab.py
tests/unit/plugins/modules/source_control/gitlab/test_gitlab_project.py

### CODE REVIEW

**Comment by Akasurde** [comment]
Migrated from https://github.com/ansible/ansible/pull/66813 

I would like to @benpatterson for this contribution. This PR had 2 ship it still it fell through the cracks.

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 1]
```diff
@@ -108,22 +116,23 @@
...
     snippets_enabled: True
     import_url: http://git.example.com/example/lab.git
     state: present
   delegate_to: localhost
 '''
 
-RETURN = '''
+RETURN = r'''
 msg:
   description: Success or failure message
```
Comment (Andersson007) [inline plugins/modules/source_control/gitlab/gitlab_project.py:137]:
```suggestion
  description: Success or failure message.
```
according to https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#return-block

File: plugins/modules/source_control/gitlab/gitlab_project.py [hunk 2]
```diff
@@ -95,11 +95,19 @@
     default: present
     type: str
     choices: ["present", "absent"]
+  merge_method:
+    description:
+      - What requirements are placed upon merges.
+      - Possible values are C(merge), C(rebase_merge) Merge commit with semi-linear history, C(ff) fast-forward merges only.
```
Comment (Andersson007) [inline plugins/modules/source_control/gitlab/gitlab_project.py:101]:
```suggestion
      - Possible values are C(merge), C(rebase_merge) merge commit with semi-linear history, C(ff) fast-forward merges only.
```

**Comment by Akasurde** [comment]
@benpatterson Thanks for the contribution. @Andersson007 Thanks for the review.

**Pull request merged**